slides/aml-08-021218-imputation-feature-selection/index.html

```diff
@@ -0,0 +1,818 @@
+      }
+      .remark-slide-content .smaller p, .remark-slide-content .smaller li,
+      .remark-slide-content .smaller .remark-code, .remark-slide-content .smaller a{
+          font-size: 25px;
+      }
+      .normal {
+          font-size: 30px;
+      }
+      .quote_author {
+          display: block;
+          text-align: right;
+          margin-top: 20px;
+          font-size: 30px;
+          font-family: 'Garamond';
+      }
+      .larger, .larger .remark-code {
+          font-size: 40px;
+      }
+      .largest, .largest .remark-code {
+          font-size: 50px;
+      }
+      .left-column, .right-column {
+          width: 48%;
+      }
+      .right-column{
+          float: right;
+      }
+      .left-column{
+          float: left;
+      }
+      .narrow-right-column {
+          float: right;
+          width: 32%
+      }
+      .wide-left-column {
+          float: left;
+          width: 65%
+      }
+      .invisible {
+          visibility: hidden
+      }
+      .remark-code, .remark-inline-code { font-family: 'Ubuntu Mono'; }
+    </style>
+  </head>
+  <body>
+    <textarea id="source">
+
+class: center, middle
+
+### W4995 Applied Machine Learning
+
+# Imputation and Feature Selection
+
+02/03/17
+
+Andreas C. Müller
+
+???
+N/A
+
+---
+class: spacious
+
+# Dealing with missing values
+
+- Missing values can be encoded in many ways
+- Numpy has no standard format for it (often np.NaN)
+- Sometimes: 999, ???, ?, np.inf, “N/A”, “Unknown“ …
+- Not discussing “missing output” - that’s semi-supervised learning.
+- Add feature that encodes that data was missing!
+- Often missingness is informative!
+
+???
+
+---
+
+.center[
+![:scale 70%](images/img_1.png)
+]
+
+- Problem : Prediction not for all samples !
+- Accuracy might be biased
+
+???
+
+---
+
+.center[
+![:scale 100%](images/img_2.png)
+]
+
+???
+
+---
+class: spacious
+
+# Imputation Methods
+
+- Mean / Median
+- kNN
+- Model-driven
+- Iterative
+
+???
+
+---
+
+
+```python
+
+from sklearn.linear_model import LogisticRegressionCV
+from sklearn.model_selection import train_test_split, cross_val_score
+X_train, X_test, y_train, y_test = train_test_split(X_, y, stratify=y, random_state=0)
+```
+```python
+nan_columns = np.any(np.isnan(X_train), axis=0)
+X_drop_columns = X_train[:, ~nan_columns]
+scores = cross_val_score(LogisticRegressionCV(v=5), X_drop_columns, y_train, cv=10)
+np.mean(scores)
+```
+```python
+0.77166672
+
+```
+
+???
+
+---
+
+# Mean and Median
+
+.center[
+![:scale 100%](images/img_4.png)
+]
+
+???
+
+---
+
+.center[
+![:scale 100%](images/img_5.png)
+]
+
+???
+
+
+---
+
+
+
+```python
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScalar
+
+nan_columns = np.any(np.isnan(X_train), axis = 0)
+X_drop_columns = X_train[:,~nan_columns]
+logreg = make_pipeline(StandardScalar(), LogisticRegression())
+scores = cross_val_score(logreg, X_drop_columns, y_train, cv = 10)
+np.mean(scores)
+```
+```python
```

Comment:
This shouldn't have "python"

Reply:
The outputs are usually just black. Also, that will remove the gray background and more clearly distinguish code and output